freqtrade/freqtradebot.py

```diff
@@ -1238,16 +1238,26 @@ def execute_trade_exit(
         return True
 
     def _notify_exit(self, trade: Trade, order_type: str, fill: bool = False,
-                     sub_trade: bool = False) -> None:
+                     sub_trade: bool = False, order: Dict = None) -> None:
         """
         Sends rpc notification when a sell occurred.
         """
-        profit_rate = trade.close_rate if trade.close_rate else trade.close_rate_requested
-        profit_trade = trade.calc_profit(rate=profit_rate)
         # Use cached rates here - it was updated seconds ago.
         current_rate = self.exchange.get_rate(
             trade.pair, refresh=False, side="sell") if not fill else None
-        profit_ratio = trade.calc_profit_ratio(profit_rate)
+        if order:
+            profit_rate = safe_value_fallback(order, 'average', 'price')
+            amount = safe_value_fallback(order, 'filled', 'amount')
+            profit = trade.process_sell_sub_trade(order, is_closed=False)
+            open_rate = trade.get_open_rate(profit, profit_rate, amount)
+            open_cost=open_rate * amount * (1+ trade.fee_open)
+            profit_ratio =  ( open_cost + profit)/open_cost - 1
```

Comment:
Well let me be very clear - NO.

We will not distribute order-parsing throughout the bot - especially not if it's just for sending notifications.

The values you require here will also be needed by other RPC methods (like /profit) - therefore they must be part of the trade object.

Reply:
its to track last buy order

Reply:
The profit is subtracted from the stake_amount. We are not taking profit unless it is a full sell.

Reply:
> We are not taking profit unless it is a full sell.

the moment the first sell order fills, this is "taken profit".
If that's for a full buy - or a partial buy is not relevant - as such, i don't understand your statement - and think that the "taken profit" needs to be captured properly in the database (and notifications should then use the data from there).

Reply:
The profit is used to adjust the position of the trade like when rebought

Reply:
please
a) make sure to merge develop into your branch. It's currently conflicting which automatically implies there must be changes done.
b) please do not use the order dict in models.py (outside of the Order object). Latest changes in develop use the `Order` object - which gives you type safety. Your code would currently crash/cause problems on certain exchanges (you can't rely on average to be filled correctly on all exchanges).